src/components/image-upload/image-upload.tsx

```diff
@@ -0,0 +1,314 @@
+  image: false,
+  title: false,
+  author: false
+};
+
+function ImageUpload({ openTrigger, entityType, entityId, user: authStatus }: TImageUploadProps) {
+  const router = useRouter();
+  const { user, token } = authStatus;
+  const [logDialogIsOpen, setLogDialogIsOpen] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [error, setError] = useState<string>();
+  const [percentage, setPercentage] = useState<number>(0);
+  const [authorCheckbox, setAuthorCheckbox] = useState(user ? true : false);
+  const [formError, setFormError] = useState<TFormErrors>(INITIAL_ERROR_STATE);
+  const imageInput = useRef<HTMLInputElement>(null);
+  const titleRef = useRef<HTMLInputElement>(null);
+  const formRef = useRef<HTMLFormElement>(null);
+
+  const initialFormState = {
+    image: null,
+    title: "",
+    author: user !== undefined ? user.fullName : "",
+    entityId: entityId,
+    entityType: entityType
+  };
+
+  const [formData, setFormData] = useState<TFormData>(initialFormState)
+  const [pickedImage, setPickedImage] = useState<string>();
+  const [pickedImageMetadata, setPickedImageMetadata] = useState<TImageMetdata>({
+    size: '',
+    name: ''
+  });
+
+  const handleImageChanged = (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.target && event.target.files && event.target.files.length > 0) {
+      const file: File = event.target.files[0];
+      if (!file) {
+        setPickedImage(undefined);
+        return;
+      }
+      setFormData({ ...formData, image: file });
+      setPickedImageMetadata({ size: bytesToSize(file.size), name: file.name });
+
+      setFormError(prevState => {
+        return {
+          ...prevState,
+          image: false
+        }
+      });
+      const fileReader: FileReader = new FileReader();
+      fileReader.onload = () => {
+        if (fileReader.result) {
+          setPickedImage(fileReader.result as string);
+        }
+      };
+      fileReader.readAsDataURL(file);
+    }
+  }
+  const handlePickClick = () => {
+    if (imageInput.current !== null) {
+      imageInput.current.click();
+    }
+  }
+
+  const handleClose = () => {
+    setIsSubmitting(true);
+  };
+
+  const resetForm = () => {
+    if (formRef.current !== null) {
+      formRef.current.reset();
+    }
+    setPickedImage(undefined);
+    setAuthorCheckbox(true);
+    setFormData(initialFormState);
+    setFormError(INITIAL_ERROR_STATE);
+  };
+
+  const handleAuthorCheckboxClick = (value: boolean): void => {
+    setAuthorCheckbox(value);
+    if (user !== undefined) {
+      setFormData({ ...formData, author: user?.fullName });
+      setFormError((prevState) => {
+        const newState = {
+          ...prevState,
+          author: false
+        };
+        return newState;
+      });
+    }
+
+  }
+
+  const onFileUploadProgressChanged = (percentage: number) => {
+    setPercentage(percentage);
+  }
+
+  useEffect(() => setError(""), [logDialogIsOpen]);
```

Comment:
Could this rather be part of resetForm?

Reply:
No. After error is set (e.q. in case of a backend errror) resetForm is called. And if error would be set to "" than user would not be able to see error. Plus, this is also used to remove any previous error that was set when dialog was opened and then closed.

Reply:
Yes, moved to resetForm.